### DISCUSSION TITLE
Error with relative path to server root not being resolved in .scss url()

### DISCUSSION THREAD

**Discussion by r-saraiva** [webpack/webpack#18068] [Q&A] [answered]
(Question on StackOverflow: https://stackoverflow.com/questions/77957211/webpack-error-module-not-found-error-cant-resolve-path-to-file-png )

Look at the following *_url.scss* file I have:
```
$host-name: 'https://my-dev-host.com';

$icon-arrow-up-down: $host-name + '/folder/img/upDownArrows.png';
```

Webpack can compile with the file above and works perfectly.

However, it poses a question to a webpack newbie like me: it doesn't make sense to update the variable `$host-name` and compile the files for every environment I have. Good sense tells me that I would be able to serve files from root, like this:

```
//$host-name: ''; --no more needed

$icon-arrow-up-down: '/folder/img/upDownArrows.png';
```

But when I do this, webpack compiler throws the following error:
"Module not found: Error: Can't resolve '/folder/img/upDownArrows.png'"

I really think this should be related with some missing config, but I was not able to figure out which.
Also, is not an option to serve the .png file from within the project.

Can you help me out?

My webpack.common.js
```
const path = require('path');
const webpack = require('webpack');
const package = require('./package.json');
const {
	CleanWebpackPlugin
} = require('clean-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const outputPath = path.join(__dirname, 'dist');
const isProduction = process.env.NODE_ENV === 'production ';

module.exports = {
	entry: {
		app: path.resolve(__dirname, './src/app.js'),
		styleguide: path.resolve(__dirname, './src/styleguide/scripts.js'),
		backoffice: path.resolve(__dirname, './src/backoffice/scripts.js'),
	},
	output: {
		path: outputPath,
		publicPath: '/dist/',
	},
	plugins: [
		new CleanWebpackPlugin({
			cleanOnceBeforeBuildPatterns: [`${outputPath}/*.hot-update.*`],
			dangerouslyAllowCleanPatternsOutsideProject: true,
			dry: false,
		}),
		new webpack.BannerPlugin({
			banner: `[base] || Version: ${package.version} || Generated: ${new Date()}`,
		}),
	],
	resolve: {
		fallback: {
			fs: false,
		}
	},
	module: {
		rules: [
			{
				test: /\.js?$/,
				exclude: /node_modules/,
			},
			{
				test: /\.s?[ac]ss$/,
				use: [
					MiniCssExtractPlugin.loader,
					{
						loader: 'css-loader',
						options: {
							importLoaders: 2,
							sourceMap: !isProduction,
						},
					},
					{
						loader: 'postcss-loader',
						options: {
							sourceMap: !isProduction,
						},
					},
					{
						loader: 'sass-loader',
						options: {
							sourceMap: !isProduction,
						},
					},
				],
			},
			{
				test: /\.(png|jpg|jpeg|svg|gif)$/,
				type: 'asset/resource',
			},
			{
				test: /\.(woff|woff2|eot|ttf|otf)$/,
				type: 'asset/resource',
			},
		],
	},
	stats: {
		errorDetails: true,
	},
};
```

If you need more information, let me know.

Thanks in advance!

**Comment by alexander-akait** [comment]
Are you sure do you have a valid URL using `$icon-arrow-up-down: '/folder/img/upDownArrows.png';`?

**Reply by r-saraiva** [reply, marked as answer]
Hello @alexander-akait !

Yes, the URL is valid.
I have an "uncommon" setup (let's say this way).
I'm using webpack just for development. Once my code is ready to release, I copy the final generated /dist/myfile.css to another tool (on another server) where all these URL's resolve correctly.

By the way, I already sorted my issue by adding the option `url: false` to the css-loader. Like this:
```
[...]
{
    loader: 'css-loader',
    options: {
      url: false,
      importLoaders: 2,
      sourceMap: !isProduction,
    },
},
[...]
```

It serves my purpose. :)
